### PR TITLE
update rhel.Dockerfile to use UBI based images

### PR DESCRIPTION
**Pull request by nickboldt**
replace quay.io/openshiftio/rhel-base-golang:latest with UBI minimal based image that builds with go-toolset-rhel7 builder

Change-Id: Iedc3e9e40e517cf1bf22ab0d85c685f71a0e0ce1
Signed-off-by: nickboldt <nboldt@redhat.com>

### CODE REVIEW

File: docker/rhel.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+RUN adduser appuser && \
+    GOOS=linux go build -o ./bin/kubernetes-image-puller ./cmd/main.go
+
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
+FROM ubi8-minimal:8.1-398
+USER root
+# CRW-528 copy actual cert
+COPY --from=builder /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem /etc/pki/ca-trust/extracted/pem/
+# CRW-528 copy symlink to the above cert
+COPY --from=builder /etc/pki/tls/certs/ca-bundle.crt                  /etc/pki/tls/certs/
```
Comment (amisevsk) [inline docker/rhel.Dockerfile:31]: Weird whitespace

File: docker/rhel.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# TODO need at least these ENV vars to be set when using the image puller; what defaults should we set in the container? 
+# export IMAGES=?
+# export IMPERSONATE_USERS=?
+# export SERVICE_ACCOUNT_ID=?
+# export SERVICE_ACCOUNT_SECRET=?
+# export OIDC_PROVIDER=?
+# export OPENSHIFT_PROXY_URL=?
+# export CACHING_INTERVAL_HOURS=?
+# export KUBERNETES_SERVICE_HOST=?
+# export KUBERNETES_SERVICE_POST=?
```
Comment (amisevsk) [inline docker/rhel.Dockerfile]: These should be set via configmap.

**Review by amisevsk — APPROVED** [review]

File: docker/rhel.Dockerfile (hunk 1 above)
Comment (nickboldt) [reply docker/rhel.Dockerfile:31]: nu uh, the target folders line up in a column. #forscience!

File: docker/rhel.Dockerfile (hunk 2 above)
Comment (nickboldt) [reply docker/rhel.Dockerfile]: we got docs on that somewhere?

Comment (nickboldt) [reply docker/rhel.Dockerfile]: or like, an example file, other than https://github.com/redhat-developer/kubernetes-image-puller#configuration ?

File: docker/rhel.Dockerfile (hunk 1 above)
Comment (amisevsk) [reply docker/rhel.Dockerfile:31]: Fair enough :)

File: docker/rhel.Dockerfile (hunk 2 above)
Comment (amisevsk) [reply docker/rhel.Dockerfile]: Most of them: https://github.com/redhat-developer/kubernetes-image-puller/blob/master/deploy/configmap.yaml

**Comment by nickboldt** [comment]
one more commit https://github.com/redhat-developer/kubernetes-image-puller/pull/22/commits/8d38e7ef29d5998f05fb1fdae55a9890f119463a

**Pull request merged**